code/codegen/src/main/kotlin/com/android/gradle/replicator/resgen/util/VectorDrawableGenerator.kt

```diff
@@ -0,0 +1,106 @@
+                    pathData = genStraightLinePathData(width, height),
+                    strokeColor = genHex(numberOfDigits = 8, random = random),
+                    strokeWidth = (random.nextInt(1000) / 1000.0).toString(),
+                    fillAlpha = (random.nextInt(1000) / 1000.0).toString()
+            ))
+        }
+        lines.add(endImage())
+        return lines
+    }
+
+    private fun startImage(width: Int, height: Int, viewportWidth: Int, viewportHeight: Int): List<String> {
+        return listOf(
+            """<?xml version="1.0" encoding="utf-8"?>""",
+            """<vector xmlns:android="http://schemas.android.com/apk/res/android"""",
+            """    android:width="${width}dp"""",
+            """    android:height="${height}dp"""",
+            """    android:viewportWidth="$viewportWidth"""",
+            """    android:viewportHeight="$viewportHeight">"""
+        )
+    }
+
+    /* TODO: add other kinds of strokes, ex:
+     *<path android:pathData="M31,63.928c0,0 6.4,-11 12.1,-13.1c7.2,-2.6 26,-1.4 26,-1.4l38.1,38.1L107,108.928l-32,-1L31,63.928z">
+     *    <aapt:attr name="android:fillColor">
+     *        <gradient
+     *            android:endX="85.84757"
+     *            android:endY="92.4963"
+     *            android:startX="42.9492"
+     *            android:startY="49.59793"
+     *            android:type="linear">
+     *            <item
+     *                android:color="#44000000"
+     *                android:offset="0.0" />
+     *            <item
+     *                android:color="#00000000"
+     *                android:offset="1.0" />
+     *        </gradient>
+     *    </aapt:attr>
+     *</path>
+     */
+    private fun path(
+            fillColor: String,
+            pathData: String,
+            strokeColor: String = "",
+            strokeWidth: String = "",
+            fillAlpha: String = ""): List<String> {
+        val lines = mutableListOf<String>()
+
+        lines.addAll(listOf(
+                """    <path""",
+                """        android:fillColor="#$fillColor""""
+        ))
+
+        if (strokeColor.isNotEmpty()) {
+            lines.add("""        android:strokeColor="#$strokeColor"""")
+        }
+
+        if (strokeWidth.isNotEmpty()) {
+            lines.add("""        android:strokeWidth="$strokeWidth"""")
+        }
+
+        if (fillAlpha.isNotEmpty()) {
+            lines.add("""        android:fillAlpha="$fillAlpha"""")
+        }
+
+        lines.add("""        android:pathData="$pathData" />""")
+
+        return lines
+    }
+
+    private fun endImage(): String {
+        return "</vector>"
+    }
+
+    private fun genStraightLinePathData(width: Int, height: Int): String {
+        val fromVector = arrayListOf(random.nextInt(width), random.nextInt(height))
+        val toVector = arrayListOf(random.nextInt(width), random.nextInt(height))
+        return "M${fromVector[0]},${fromVector[1]}L${toVector[0]},${toVector[1]}"
```

Comment:
why having fromVector and toVector, this can be simplified with just 
private fun genStraightLinePathData(width: Int, height: Int): String = 
     "M${random.nextInt(width)},...."

Reply:
I thought this would make it more obvious what the line was, but I guess I can turn that into a comment.